app/models/ebean/AlertExtension.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2015 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package models.ebean;
+
+import com.avaje.ebean.Model;
+import com.avaje.ebean.annotation.CreatedTimestamp;
+import com.avaje.ebean.annotation.UpdatedTimestamp;
+
+import java.sql.Timestamp;
+import javax.persistence.Column;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.OneToOne;
+import javax.persistence.Version;
+
+/**
+ * Abstract model class for extensions data for alerts.
+ *
+ * @author Deepika Misra (deepika at groupon dot com)
+ */
+@MappedSuperclass
+public abstract class AlertExtension extends Model {
```

Comment:
Other than the meta-data boilerplate which is effectively in all our models, the only piece in here that seems reusable by other extension implementations is the alert field (which is the fk; right?). So my question is what is the purpose of the abstract base class? Using inheritance only for this type of code reuse is not always beneficial. Do we foresee this base class having more in it than just the meta-data?